Added Horse Types To Horse.java.

ewlsh opened the pull request:
This doesn't add the 35 different combinations of NORMAL Horses, but it would allow us to set the type of the horse... Source For Types: http://www.minecraftwiki.net/wiki/Item_id#Horse_Variants

Commit:
Added Horse Types To Horse.java.

This doesn't add the 35 different combinations of NORMAL Horses, but it would allow us to set the type of the horse... Source For TypeS: http://www.minecraftwiki.net/wiki/Item_id#Horse_Variants

turt2live commented on src/main/java/org/bukkit/entity/Horse.java:
```diff
@@ -3,4 +3,44 @@
-public interface Horse extends Animals, Vehicle {}
+public interface Horse extends Animals, Vehicle {
+
+    public Type getHorseType();
```
Inconsistent return value

Edit: and extra lines below

turt2live commented on src/main/java/org/bukkit/entity/Horse.java:
```diff
@@ -3,4 +3,44 @@
+            return (id >= types.length) ? null : types[id];
+        }
+    }
+  
```
More extra lines

turt2live commented on src/main/java/org/bukkit/entity/Horse.java:
```diff
@@ -3,4 +3,44 @@
+        public int getId() {
+            return id;
+        }
+
```
Extra lines

turt2live commented on src/main/java/org/bukkit/entity/Horse.java:
```diff
@@ -3,4 +3,44 @@
+        private HorseType(int id) {
+            this.id = id;
+        }
+
```
~~Extra lines~~

Edit: I'm an idiot.

turt2live commented on src/main/java/org/bukkit/entity/Horse.java:
```diff
@@ -3,4 +3,44 @@
  * Represents a Horse.
  */
-public interface Horse extends Animals, Vehicle {}
+public interface Horse extends Animals, Vehicle {
```
Your newly added methods & enum need JavaDocs

Commit:
Update Horse.java

Fixed Error on Line 9 (Typed "Type" instead of "HorseType")
Fixed Error on Line 19 (Typed ";" instead of ","

Commit:
Update Horse.java

Added Comments...

ewlsh commented:
I think I added what you wanted :P

turt2live commented on src/main/java/org/bukkit/entity/Horse.java:
```diff
@@ -3,4 +3,66 @@
+     */
+    public void setHorseType(HorseType type);
+
+     /*
```
This is a block comment, not JavaDocs

turt2live commented:
Don't forget the whitespace. A bukkit member will find more than what I have though (At least they usually do)

Commit:
Update Horse.java

There...

ewlsh commented:
Okay... I checked this against Skeleton.java and Ocelot.java and they now mirror each other except for the Horse part, so I don't think anything is wrong anymore.

turt2live commented:
You still have excess lines

Commit:
Update Horse.java

Removed "Extra Lines"

ewlsh commented:
Good?

turt2live commented:
Still have a few

Commit:
Update Horse.java

....

ewlsh commented:
The spacing now matches Skeleton.java ...I found 1 extra space :|

turt2live commented:
Still one extra line... I have it commented on the diff. 

And don't match to another file in Bukkit, match to the [guidelines](https://github.com/Bukkit/Bukkit/blob/master/CONTRIBUTING.md).

ewlsh commented:
Can you just give me the line number :|

turt2live commented:
43, as shown by previous comments.

ewlsh commented:
Ah yes... I found it, but considering that in the other Bukkit classes that there is a space there... I don't see any reason to remove it :|

ewlsh commented:
https://github.com/Bukkit/Bukkit/blob/master/src/main/java/org/bukkit/entity/Skeleton.java
- Line 40

turt2live commented:
Oh, I'm an idiot. I for some reason saw 2 lines there...

Ignore that... 

Looks good, all you have to do now is wait :)

ewlsh commented:
:P

Black-Hole commented:
Since this pull request doesn't follow the guidelines for a PR and lacks the implementation part in CraftBukkit, there is no chance this will get pulled.
But I think at some time before the first beta release of 1.6.x Bukkit will come up with a complete solution.
At least you learned some of the steps for a pull request. :)

drtshock commented:
Follow the contributing guidelines :)

https://github.com/Bukkit/Bukkit/blob/master/CONTRIBUTING.md

riking commented on src/main/java/org/bukkit/entity/Horse.java:
```diff
@@ -3,4 +3,61 @@
+         *
+         * @return Horse type ID
+         */
+        public int getId() {
```
I think the mapping of enum -> NMS ID should be done craftbukkit-side.

mbax commented:
Handled in https://github.com/Bukkit/Bukkit/commit/1a8fb592da70831c3952c7dc8bea9eee0a8527a8

Thanks for the PR :)
